sphinx_gallery/tests/test_gen_gallery.py

```diff
@@ -166,6 +169,23 @@ def test_config_backreferences(config_app):
     assert build_warn == ""
 
 
+@pytest.mark.skipif(sys.version_info[:2] == (3, 4),
+                    reason="Won't work on Python 3.4")
+def test_duplicate_files_warn(config_app):
+    """Test for a warning when two files with the same filename exist."""
+    files = ['./a/file1.py', './a/file2.py', './b/file1.py']
+    msg = ("Duplicate file name(s) found. Having duplicate file names will "
+           "break some links. List of files: %s")
+    m = "['./b/file1.py']" if sys.version_info[0] >= 3 else "[u'./b/file1.py']"
+    with LogCapture() as log:
+        check_duplicate_filenames(files)
+        log.check(('sphinx-gallery', 'WARNING', msg % m))
```

Comment:
If you use `format` instead of the `%` operator you should be able to skip the problem of defining the unicode for `m`. Other option is to `from __future__ import unicode_literals`

Reply:
I'm using % because I thought that's what @larsoner prefers...if y'all are fine with `format` then I'm happy to change it. @larsoner do you have opinions?

Reply:
`%` is a local MNE pref, SG tends to use `.format` from what I've seen